l10n_co_res_partner/models/l10n_co_res_partner.py

```diff
@@ -0,0 +1,459 @@
+            (12, "12 - Identity Card"),
+            (13, "13 - Citizenship Card"),
+            (21, "21 - Alien Registration Card"),
+            (22, "22 - Foreigner ID"),
+            (31, "31 - TAX Number (NIT)"),
+            (41, "41 - Passport"),
+            (42, "42 - Foreign Identification Document"),
+            (43, "43 - No Foreign Identification")
+
+        ], "Type of Identification"
+    )
+    xidentification = fields.Char("Document Number", store=True,
+                                  help="Enter the Identification Number")
+    verificationDigit = fields.Integer('VD', size=2)
+    formatedNit = fields.Char(
+        string='NIT Formatted',
+        compute="_compute_concat_nit",
+        store=True
+    )
+
+    # Tributate regime
+    pn_retri = fields.Selection(
+        [
+            (6, "Simplified"),
+            (23, "Natural Person"),
+            (7, "Common"),
+            (11, "Great Taxpayer Autorretenedor"),
+            (22, "International"),
+            (25, "Common Autorretenedor"),
+            (24, "Great Contributor")
+        ], "Tax Regime"
+
+    )
+
+    # CIIU - Clasificación Internacional Industrial Uniforme
+    ciiu = fields.Many2one('ciiu', "ISIC Activity")
+    personType = fields.Selection(
+        [
+            (1, "Natural"),
+            (2, "Juridical")
+        ],
+        "Type of Person",
+        default=1
+    )
+
+    # Replacing the field company_type
+    company_type = fields.Selection(
+        [
+            ('person', 'Individual'),
+            ('company', 'Company')
+        ]
+    )
+
+    # Boolean if contact is a company or an individual
+    is_company = fields.Boolean(string=None)
+
+    # Verification digit
+    dv = fields.Integer(string=None, store=True)
+
+    # Country -> State -> Municipality - Logic
+    country_id = fields.Many2one('res.country', "Country")
+    xcity = fields.Many2one('res.country.state.city', "Municipality")
+    city = fields.Char(related="xcity.name")
+
+    # identification field has to be unique,
+    # therefore a constraint will validate it:
+    _sql_constraints = [
+        ('ident_unique',
+         'UNIQUE(doctype,xidentification)',
+         "Identification number must be unique!"),
+    ]
+
+    # Check to handle change of Country, City and Municipality
+    change_country = fields.Boolean(string="Change Country / Department?",
+                                    default=True, store=False)
+
+    # Name of point of sales / delivery contact
+    pos_name = fields.Char("Point of Sales Name")
+
+    # Birthday of the contact (only useful for non-company contacts)
+    xbirthday = fields.Date("Birthday")
+
+    @api.depends('xidentification')
+    @api.multi
+    def _compute_concat_nit(self):
+        """
+        Concatenating and formatting the NIT number in order to have it
+        consistent everywhere where it is needed
+        @return: void
+        """
+        # Executing only for Document Type 31 (NIT)
```

Comment:
`self`is a recordset, not a single record.
You need to putt all the code inside a loop: `for partner in self:`